src/organizations/OrganizationEntity.js

```diff
@@ -217,7 +240,32 @@ export class OrganizationEntity extends Entity {
 		if (!this._entity || !this._entity.hasLinkByRel(rel)) {
 			return;
 		}
-
 		return this._entity.getLinkByRel(rel).href;
 	}
+
+	_getCompletionAction(enabled) {
+		const actionName = enabled ? actions.trackCompletion : actions.doNotTrackCompeltion;
+		return this._entity.getActionByName(actionName);
+	}
+
+	_getDisplayAction(enabled) {
+		const actionName = enabled ? actions.displayProgress : actions.doNotDisplayProgress;
+		return this._entity.getActionByName(actionName);
+	}
+
+	updateCompletionTracking(enabled) {
+		const action = this._getCompletionAction(enabled);
+		if (action) {
+			return performSirenAction(this._token, action, action.fields, false);
+		}
+		return Promise.resolve(undefined);
+	}
+
+	updateDisplayProgress(enabled) {
```

Comment:
If you make this async you don't need to do `return Promise.resolve(undefined);`